controllers/oneagent/oneagent_controller.go

```diff
@@ -128,20 +125,9 @@ func (r *ReconcileOneAgent) reconcileRollout(ctx context.Context, rec *utils.Rec
 	}
 
 	// Check if this DaemonSet already exists
-	dsActual := &appsv1.DaemonSet{}
-	err = r.client.Get(ctx, types.NamespacedName{Name: dsDesired.Name, Namespace: dsDesired.Namespace}, dsActual)
-	if err != nil && k8serrors.IsNotFound(err) {
-		rec.Log.Info("Creating new daemonset")
-		if err = r.client.Create(ctx, dsDesired); err != nil {
-			return false, err
-		}
-	} else if err != nil {
-		return false, err
-	} else if hasDaemonSetChanged(dsDesired, dsActual) {
-		rec.Log.Info("Updating existing daemonset")
-		if err = r.client.Update(ctx, dsDesired); err != nil {
-			return false, err
-		}
+	updateCR, err = kubeobjects.CreateOrUpdateDaemonSet(r.client, r.logger, dsDesired)
+	if err != nil {
+		return updateCR, err
```

Comment:
this refactoring makes the comment in line 127 wrong, please remove the old comment.
I really like how this refactoring turned out though